### PR TITLE
[VK] Fix playlist URLs

### LINKED ISSUE #2825: VK.com Playlists can't be downloaded anymore
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.02.04**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### Description

A few months ago vk.com changed their url schema. For a while you were able to convert the new url into the old url, but that is not possible anymore.

example new url: https://vk.com/video/playlist/-28532279_55889438
example old url: https://vk.com/videos-28532279?section=album_55889438

I've found a similar [issue](https://github.com/yt-dlp/yt-dlp/issues/2196) but that only concerns channels (even though it mentions playlists in the title).

### Verbose log

```shell
[debug] Command-line config: ['-vU', '-i', 'https://vk.com/video/playlist/-28532279_55889438']
[debug] Home config "yt-dlp.conf": ['-o', '~/Movies/PRIVATE/VR/%(title)s/%(title)s-%(id)s.%(ext)s', '--download-archive', '~/.config/youtube-dl/downloaded.txt']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8 (No ANSI), err utf-8, pref UTF-8
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Python version 3.9.10 (CPython 64bit) - macOS-11.4-x86_64-i386-64bit
[debug] exe versions: ffmpeg 5.0 (setts), ffprobe 5.0
[debug] Optional libraries: sqlite
[debug] Proxy map: {}
[debug] Loading archive file '/Users/PRIVATE/.config/youtube-dl/downloaded.txt'
[debug] [generic] Extracting URL: https://vk.com/video/playlist/-28532279_55889438
WARNING: [generic] Could not send HEAD request to https://vk.com/video/playlist/-28532279_55889438: HTTP Error 418: 
WARNING: [generic] Falling back on generic information extractor.
[debug] Looking for video embeds
[debug] [generic] Extracting URL: https://vk.com/badbrowser.php
WARNING: [generic] Could not send HEAD request to https://vk.com/badbrowser.php: HTTP Error 418: 
WARNING: [generic] Falling back on generic information extractor.
[debug] Looking for video embeds
ERROR: Unsupported URL: https://vk.com/badbrowser.php
Traceback (most recent call last):
  File "/usr/local/Cellar/yt-dlp/2022.2.4/libexec/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1381, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/Cellar/yt-dlp/2022.2.4/libexec/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1451, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/Cellar/yt-dlp/2022.2.4/libexec/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 612, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/Cellar/yt-dlp/2022.2.4/libexec/lib/python3.9/site-packages/yt_dlp/extractor/generic.py", line 3986, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://vk.com/badbrowser.php
```

### PR DESCRIPTION
**Pull request by the-marenga**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

VK changed their playlist URLs from 
`https://vk.com/videosPAGEID?section=album_PLAYLISTID` to 
`https://vk.com/video/playlist/PAGEID_PLAYLISTID`, 
which meant new URLs had to be rewritten to the old format by hand to work. 
a70b71e85ad61c5f7c85ef920d72f949fb767d53 then removed support for the old version by changing the regex to fix another URL related issue.

This PR now restores playlist downloads by correctly parsing the new URL and indirectly restores support for the old format via an automatic redirect to the new page.

Fixes #2825

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/vk.py [hunk 1]
```diff
@@ -584,11 +591,19 @@ def _entries(self, page_id, section):
...
         u_id, section = self._match_valid_url(url).groups()
         webpage = self._download_webpage(url, u_id)
-        page_id = self._search_regex(r'data-owner-id\s?=\s?"([^"]+)"', webpage, 'page_id')
+
+        if u_id.startswith("@"):
+            page_id = self._search_regex(r'data-owner-id\s?=\s?"([^"]+)"', webpage, 'page_id')
+        elif "_" in u_id:
+            page_id, section = u_id.split("_", 1)
+        else:
+            raise ExtractorError("Invalid URL", expected=True)
```
Comment (pukkandan) [inline yt_dlp/extractor/vk.py:600]: use single quotes

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Comment by BrunoKrinski** [comment]
I'm still having problem with vk playlists links. I tested with several playslistas like this https://vk.com/videos-28532279?section=album_55889438 and this http://vk.com/video/playlist/-174476437_2 and it returns this error: ERROR: Unsupported URL: https://vk.com/badbrowser.php

**Comment by BrunoKrinski** [comment]
Hi, thanks for answering me. I installed with pip. The version installed is 2022.10.04. And Python 3.9. It seems that it don't recognize as a vk link. It fall on Generic extractor: [generic] -174476437_2: Downloading webpage and then this error: ERROR: Unsupported URL: https://vk.com/badbrowser.php.

**Comment by the-marenga** [comment]
The current version (2022.10.04) does not include this fix here yet. You either have to wait (~2-3 weeks?) for the new version to include it, or build yt-dlp from source

**Comment by BrunoKrinski** [comment]
ok. I'll wait then. Thanks

**Comment by BrunoKrinski** [comment]
Hi, sorry to bother again. Now I'm having another problem when downloading vk playslists. I can't continue or start new download from a playlist. Tried different playlists and all have the same problem.

[vk:uservideos] Extracting URL: https://vk.com/video/playlist/261475876_2
[vk:uservideos] 261475876_2: Downloading webpage
[download] Downloading playlist: vk3
[vk:uservideos] 261475876: Downloading JSON metadata
ERROR: Unknown error
[vk:uservideos] Playlist vk3: Downloading 1 items of 0
[download] Finished downloading playlist: vk3

**Comment by the-marenga** [comment]
Yeah, their API changed again. The fix is pretty simple, so it should probably not take long to get merged. 

In the meantime you should be able to somewhat fix it yourself by adding a `section=playlist_` before the last number in your url. 

https://vk.com/video/playlist/261475876_2

becomes 

https://vk.com/video/playlist/261475876_section=playlist_2

You playlist is private, so I cant test it, but it should work. If not, you have to just wait for the real fix: https://github.com/yt-dlp/yt-dlp/pull/6122

**Pull request merged**